python/ray/tune/ray_trial_executor.py

```diff
@@ -906,6 +761,111 @@ def _change_working_directory(self, trial):
         else:
             yield
 
+    def get_next_future_result(self, next_trial_exists: bool) -> FutureResult:
```

Comment:
Would it be possible to break this function up a little? I feel it's hard to read